### PR TITLE
Change certificate setup

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

**add description**

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- https://github.com/uyuni-project/uyuni-docs/pull/1345

- [ ] **DONE**

## Test coverage
- No tests: already covered

- [x] **DONE**

## Links

Fixes #
Tracks # **add downstream PR, if any**

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

https://ci.nue.suse.com/view/Manager/view/Uyuni-PRs/job/uyuni-prs-ci-tests/955/

### CODE REVIEW

File: proxy/proxy/spacewalk-proxy.changes [hunk 1]
```diff
@@ -1,3 +1,5 @@
+- create directoies for TLS Certificates
```
Comment (meaksh) [inline proxy/proxy/spacewalk-proxy.changes:1]: Small typo:
```suggestion
- create directories for TLS Certificates
```

File: spacewalk/certs-tools/mgr_ssl_cert_setup.py [hunk 2]
```diff
@@ -0,0 +1,465 @@
...
+import shutil
+import tempfile
+import time
+import subprocess
+from datetime import datetime
+
+from spacewalk.common.rhnLog import initLOG, log_time, log_clean
+from uyuni.common.fileutils import getUidGid
+
+logfile = '/var/log/rhn/mgr-ssl-cert-setup.log'
```
Comment (meaksh) [inline spacewalk/certs-tools/mgr_ssl_cert_setup.py:28]: Small question: isn't this a constant? Is there any reason to not define this as `LOGFILE` as the rest of constants?
```suggestion
LOGFILE = '/var/log/rhn/mgr-ssl-cert-setup.log'
```

File: spacewalk/certs-tools/mgr_ssl_cert_setup.py [hunk 3]
```diff
@@ -0,0 +1,465 @@
...
+     parser = argparse.ArgumentParser(usage=usage)
+     parser.add_argument('-r', '--root-ca-file', help='Path to the Root CA')
+     parser.add_argument('-i', '--intermediate-ca-file', action='append', default=[],
+                         help='Path to an intermediate CA')
+     parser.add_argument('-s', '--server-cert-file', help='Path to the Server Certificate')
+     parser.add_argument('-k', '--server-key-file', help='Path to the Server Private Key')
+     parser.add_argument('--verbose', '-v', action='count', default=0)
+
+     options = parser.parse_args()
+     initLOG(logfile, options.verbose or 1)
```
Comment (meaksh) [inline spacewalk/certs-tools/mgr_ssl_cert_setup.py:84]:
```suggestion
     initLOG(LOGFILE, options.verbose or 1)
```

File: spacewalk/certs-tools/mgr_ssl_cert_setup.py [hunk 4]
```diff
@@ -0,0 +1,465 @@
...
+    checkCompleteCAChain(workdir, certData)
+
+def _main():
+    """ main routine """
+
+    options = processCommandline()
+    checkOptions(options)
+
+    with tempfile.TemporaryDirectory() as workdir:
+        #import pdb; pdb.set_trace()
```
Comment (meaksh) [inline spacewalk/certs-tools/mgr_ssl_cert_setup.py:424]: I think we can drop this one line
```suggestion
```

**Review by meaksh — COMMENTED** [review]
Thanks for introducing this tool @mcalmer ! I just made some small comments

File: spacewalk/certs-tools/mgr-ssl-cert-setup [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+
+## language imports
+import sys
+import os
+
+
+mod_name = os.path.basename(sys.argv[0]).replace('-', '_')
+try:
+    mod = __import__("certs." + mod_name)
```
Comment (vzhestkov) [inline spacewalk/certs-tools/mgr-ssl-cert-setup:24]: Is it required to have such hacky import?

Comment (mcalmer) [reply spacewalk/certs-tools/mgr-ssl-cert-setup:24]: Good catch. I will remove it

Comment (vzhestkov) [reply spacewalk/certs-tools/mgr-ssl-cert-setup:24]: Additionally maybe to run black over `mgr_ssl_cert_setup.py`?

**Review by meaksh — APPROVED** [review]

**Pull request merged**